scripts/add_electricity.py

```diff
@@ -327,100 +337,99 @@ def attach_hydro(n, costs, ppl):
+                        .transpose('time', 'name')
+                        .to_pandas()
+                        .multiply(dist_key, axis=1))
+
+    if 'ror' in carriers and not ror.empty:
+        n.madd("Generator", ror.index,
+               carrier='ror',
+               bus=ror['bus'],
+               p_nom=ror['p_nom'],
+               efficiency=costs.at['ror', 'efficiency'],
+               capital_cost=costs.at['ror', 'capital_cost'],
+               weight=ror['p_nom'],
+               p_max_pu=(inflow_t[ror.index]
+                         .divide(ror['p_nom'], axis=1)
+                         .where(lambda df: df<=1., other=1.)))
+
+    if 'PHS' in carriers and not phs.empty:
+        # fill missing max hours to config value and assume no natural inflow
+        # due to lack of data
+        phs = phs.replace({'max_hours': {0: c['PHS_max_hours']}})
+        n.madd('StorageUnit', phs.index,
+               carrier='PHS',
+               bus=phs['bus'],
+               p_nom=phs['p_nom'],
+               capital_cost=costs.at['PHS', 'capital_cost'],
+               max_hours=phs['max_hours'],
+               efficiency_store=np.sqrt(costs.at['PHS','efficiency']),
+               efficiency_dispatch=np.sqrt(costs.at['PHS','efficiency']),
+               cyclic_state_of_charge=True)
+
+    if 'hydro' in carriers and not hydro.empty:
+        hydro_max_hours = c.get('hydro_max_hours')
+        if hydro_max_hours == 'energy_capacity_totals_by_country':
+            # TODO: check where the statistics come from, IE's p_nom is totally
+            # underrepresented
+            e_target = (pd.read_csv(snakemake.input.hydro_capacities,
+                              index_col=0)["E_store[TWh]"].clip(lower=0.2)*1e6)
+            e_installed = hydro.eval('p_nom * max_hours').groupby(hydro.country).sum()
+            e_missing = e_target - e_installed
+            missing_mh_i = hydro.query('max_hours == 0').index
+
+            max_hours_country = e_missing / hydro.loc[missing_mh_i].groupby('country').p_nom.sum()
+            hydro_max_hours = hydro.max_hours.where(hydro.max_hours > 0,
+                                    hydro.country.map(max_hours_country))
+        elif hydro_max_hours == 'estimate_by_large_installations':
+            hydro_capacities = pd.read_csv(snakemake.input.hydro_capacities,
+                                           comment="#", na_values='-', index_col=0)
+            estim_hydro_max_hours = hydro_capacities['E_store[TWh]'] * 1e3 / \
+                                    hydro_capacities['p_nom_discharge[GW]']
+
+            missing_countries = (pd.Index(hydro['country'].unique())
+                                 .difference(estim_hydro_max_hours.dropna().index))
+            if not missing_countries.empty:
+                logger.warning("Assuming max_hours=6 for hydro reservoirs in the countries: {}"
+                               .format(", ".join(missing_countries)))
```

Comment:
Do we still need these amazingly crude estimations, how well does just using the JRC data work out? 

`snakemake.input.hydro_capacities` contain hand researched energy storage capacities from Emil. Maybe check in his thesis for the full references: `/home/vres/data/literature/MasterThesis/150729_MasterThesis_EmilEriksen.pdf`

Reply:
I think we have to rely on them a bit longer. So far only roughly a third of all hydro dams in JRC have stated storage capacities. But then on the other hand, the question is, is it worth it to modify the option 
```
hydro_max_hours == 'estimate_by_large_installations'
```
so that non-NA `max_hours` values are untouched?

Reply:
Yes, i'd think so!